src/v/kafka/server/handlers/init_producer_id.cc

```diff
@@ -19,13 +19,31 @@
 #include "kafka/server/logger.h"
 #include "kafka/server/request_context.h"
 #include "kafka/server/response.h"
+#include "model/record.h"
 #include "utils/remote.h"
 #include "utils/to_string.h"
 
 #include <seastar/core/print.hh>
 
 namespace kafka {
 
+namespace {
+
+// Provided pid in init_producer_id request can not be {x >= 0, -1} or
+// {-1, x >= 0}.
+bool is_invalid_pid(model::producer_identity expected_pid) {
+    if (expected_pid == model::unknown_pid) {
```

Comment:
why `unknown_pid` is invalid? I was under impression that when a client doesn't use kip-360 then expected_pid isn't set so the request can't be invalid.

Reply:
oops